students/EricAdams/trapezoidal_rule/trapezoidal_rule.py

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env python
+
+
+def f(x, a=1, b=1, c=1):
+    return a * (x**2) + b * x + c
+
+
+def sigma(func, n, a, b):
+    '''
+    calculate the sigma portion of the trapezoidal rule for the area
+    under a curve
+    :func = the function that defines the curve
+    :func = function type
+    :n = the number of steps in the integration
+    :n = integer type
+    :a = the starting point on the x axis for the integration
+    :a = numeric type
+    :b = the end point on the x axis for the integration
+    :b = numeric type
+    :step = float type
+    '''
+    # value of the sigma portion of the trapez. rule
+    step = float((b - a) / n)
+    # use a listcomp to calculate each element of sigma
+    # sigma_list = [func(a + i * step) for i in range(1, n)]
+    # return sum(sigma_list)
+    # use a genexp to calculate each element of sigma
+    result = 0
+    sigma_gen = (func(a + i * step) for i in range(1, n))
+    for i in sigma_gen:
+        result += i
+    return result
+
+
+def func_a_plus_func_b_div_2(func, a, b):
```

Comment:
not much point in breaking these out to different functions -- they are one or a few lines, aren't likely to change, and don't need to be called from more than one place.

But if you do break them up, then it might make sense to put them in a callable class to keep it all in one namespace.

Or, alternatively,  define these utility function inside the main trapz_area_under_curve function, so again, one namespace.